.github/workflows/icon4py-qa.yml

```
@@ -33,4 +33,4 @@ jobs:
             - name: Run checks
               run: |
                   source .venv/bin/activate
-                  pre-commit run
+                  SKIP=mypy pre-commit run --all-files
```

There are two packages where we fixed `mypy` (see `.pre-commit-config.yml`): the stencil tests in the dycore and tools. So rather than globally skipping mypy it would be good to run it there otherwise those will rot again.

(In fact they already do, because I get some errors for the `tools` package. I suggest to do the following:
remove the skip here: 
```suggestion
                  pre-commit run --all-files
```
and fix the issues in the `tools` package. 
You can fix all of them with a `# type: ignore[<rule>]` comment. 

(All except for one are in "deprecated" code that will be deleted eventually because it is part of Fortran to GT4Py integration methods that we do not use any longer. The one that is not, already has such a comment but the wrong rule, if I see that correctly.

See patch file: 
[mypy_fixes_in_tools.txt](https://github.com/user-attachments/files/20202842/mypy_fixes_in_tools.txt)

@egparedes does that make sense?

---

Thanks @halungge, that's definitely better! I applied your diff and:

- ignored warnings locally for `icon4py.model.atmosphere.dycore.dycore_states` since there were only three of them (an empty `py.typed` file also works for this, is that nicer?)
- added the empty `py.typed` file for testing modules

With this it all the checks pass for me locally. I hope CI will agree...